test/virtual/include/openenclave/edger8r/common.h

```diff
@@ -153,6 +153,34 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
 
 #define OE_WRITE_IN_OUT_PARAM OE_WRITE_IN_PARAM
 
+#define OE_WRITE_DEEPCOPY_OUT_PARAM(argname, argsize)                          \
+    do                                                                         \
+    {                                                                          \
+        void* p = (void*)(_deepcopy_out_buffer + _deepcopy_out_buffer_offset); \
+        OE_ADD_SIZE(_deepcopy_out_buffer_offset, (size_t)(argsize));           \
+        memcpy(p, argname, (size_t)(argsize));                                 \
+    } while (0)
+
+#define OE_SET_DEEPCOPY_OUT_PARAM(argname, argsize, argtype)               \
+    if (argname)                                                           \
+    {                                                                      \
+        argname = (argtype)oe_malloc(argsize);                             \
```

Comment:
Do you expect the user to free the out parameters via `free` or `oe_free`? The memory allocation and the design doc needs to match that.

Reply:
What's the difference between the two? By looking into the implementation,
- On the enclave side: both `oe_malloc` and `malloc` internally invokes `oe_allocator_malloc`
- On the host side: `oe_malloc` is simply a wrapper of `malloc`.

The decision here is because that today we didn't expose `malloc` but `oe_malloc` to the oeedger8r generated enclave and host code.

Reply:
There is no guarantee that oe-malloc == malloc, both on the host-side and the enclave-side.
Therefore, if we want the user to free out buffers using oe-free, we ought to use oe-malloc. Instead if we want the user to use free, then we ought to use malloc for these buffers.

Reply:
Updated to use `malloc` such that user should use `free`.